Game refuses to compile now

Uh oh.  I downloaded the game this morning and recompiled, and got this.  Did we do a boo-boo?

> /home/leon/raceintospace/src/game/admin.cpp: In function ‘int SaveGame(std::vector<{anonymous}::SFInfo>)’:
/home/leon/raceintospace/src/game/admin.cpp:1942:12: error: ‘struct SaveFileHdr’ has no member named ‘compSize’
 1942 |     header.compSize = interimData.endTurnSaveSize;
      |            ^~~~~~~~
/home/leon/raceintospace/src/game/admin.cpp:1949:39: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
 1949 |         write_save_file(filename.c_str(), header);
      |                         ~~~~~~~~~~~~~~^~
      |                                       |
      |                                       const char*
/home/leon/raceintospace/src/game/admin.cpp:1812:28: note:   initializing argument 1 of ‘void write_save_file(char*, SaveFileHdr)’
 1812 | void write_save_file(char *Name, SaveFileHdr header)
      |                      ~~~~~~^~~~
/home/leon/raceintospace/src/game/admin.cpp:1951:38: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
 1951 |         write_save_file(savegames[i].Name, header);
/home/leon/raceintospace/src/game/admin.cpp:1812:28: note:   initializing argument 1 of ‘void write_save_file(char*, SaveFileHdr)’
 1812 | void write_save_file(char *Name, SaveFileHdr header)
      |                      ~~~~~~^~~~
/home/leon/raceintospace/src/game/admin.cpp:1954:1: warning: control reaches end of non-void function [-Wreturn-type]
 1954 | }
      | ^
make[2]: *** [src/game/CMakeFiles/raceintospace.dir/build.make:76: src/game/CMakeFiles/raceintospace.dir/admin.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:433: src/game/CMakeFiles/raceintospace.dir/all] Error 2
make: *** [Makefile:166: all] Error 2

![CompileError](https://user-images.githubusercontent.com/3118014/172057258-bb44bd89-4a75-499d-b00c-fa8112e5ac56.png)

--- a/src/game/admin.cpp
+++ b/src/game/admin.cpp
@@ -124,7 +124,7 @@ void LoadGame(const char *filename);
 void LegacyLoad(SaveFileHdr header, FILE *fin, size_t fileLength);
 bool OrderSaves(const SFInfo &a, const SFInfo &b);
 char RequestX(const char *s, char md);
-void write_save_file(char *Name, SaveFileHdr header);
+void write_save_file(const char *Name, SaveFileHdr header);
 int SaveGame(const std::vector<SFInfo> savegames);
 
 namespace
@@ -1809,7 +1809,7 @@ int32_t EndOfTurnSave(char *inData, int dataLen)
  * data are serialized into a JSON string, compressed by zlib, and
  * written to disk.
  */
-void write_save_file(char *Name, SaveFileHdr header)
+void write_save_file(const char *Name, SaveFileHdr header)
 {
     FILE *fin;
     int i, offset, size;
@@ -1939,7 +1939,6 @@ int SaveGame(const std::vector<SFInfo> savegames)
     i--;  // decrement to correct for the FOR loop
 
     EndOfTurnSave((char *) Data, sizeof(struct Players));
-    header.compSize = interimData.endTurnSaveSize;
 
     // Create the filename from the title.
     // The field savegames[i].Name is a filename provided by

